v56-team22-surgery-status-board/src/components/PatientStatusTable/PaginationController.tsx

```diff
@@ -0,0 +1,134 @@
+import  { useState } from 'react';
+import { type Table as ReactTable } from '@tanstack/react-table';
+import {
+  PaginationNext,
+  PaginationPrevious,
+} from '@/components/ui/pagination';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { Button } from '@/components/ui/button';
+
+import {
+  FaAngleDoubleLeft,
+  FaAngleDoubleRight,
+  FaAngleLeft,
+  FaAngleRight,
+} from 'react-icons/fa';
+
+// type PaginationMode = 'manual' | 'auto' | 'paused';
+interface PaginationControllerProps<TData> {
+  table: ReactTable<TData>;
+}
+
+function PaginationController<TData>({
+  table,
+}: PaginationControllerProps<TData>) {
+  const [pageSize, setPageSize] = useState(
+    table.getState().pagination.pageSize
+  );
+
+  const totalPages = table.getPageCount();
+  const currentPage = table.getState().pagination.pageIndex;
+  const totalItems = table.getRowModel().rows.length;
```

Comment:
I noticed on line 57, `totalItems` seems to be showing total items per page view. 
Something like  `const totalItemsFiltered = table.getFilteredRowModel().rows.length` would give the total number of rows in the table.